meson.build

```diff
@@ -0,0 +1,174 @@
+warning_disable = [
+    # Warn for variables that might be changed by longjmp or vfork. Disable because of constant false positives/negatives.
+    'clobbered',
+
+    # Warn if a structure’s initializer has some fields missing. Disable so we can initialize with {0}.
+    'missing-field-initializers',
+
+    # Warn when a switch case falls through. Disable because this an useful aspect of switches and tests should catch problems.
+    'implicit-fallthrough',
+]
+
+foreach warning : warning_disable
+    if meson.get_compiler('c').has_argument('-W' + warning)
+        add_project_arguments('-Wno-' + warning, language: 'c')
+    endif
+endforeach
+
+####################################################################################################################################
+# Enable additional optimizations if the level is high enough
+####################################################################################################################################
+if get_option('optimization') in ['2', '3']
+    optimization_enable = [
+        # Unroll loops whose number of iterations can be determined at compile time or upon entry to the loop
+        'unroll-loops',
+
+        # Perform loop vectorization on trees
+        'tree-vectorize',
+    ]
+
+    foreach optimization : optimization_enable
+        if meson.get_compiler('c').has_argument('-f' + optimization)
+            add_project_arguments('-f' + optimization, language: 'c')
+        endif
+    endforeach
+endif
+
+####################################################################################################################################
+# Stop after the first error when error on warn enabled. Subsequent errors are often caused by the first error.
+####################################################################################################################################
+if get_option('fatal-errors') and meson.get_compiler('c').has_argument('-Wfatal-errors')
+        add_project_arguments('-Wfatal-errors', language: 'c')
+endif
+
+####################################################################################################################################
+# Build configuration
+####################################################################################################################################
+configuration = configuration_data()
+
+# Find required bz2 library
+lib_bz2 = meson.get_compiler('c').find_library('bz2')
+
+# Find optional lz4 library
+lib_lz4 = dependency('liblz4', required: false)
+
+if lib_lz4.found()
+    configuration.set('HAVE_LIBLZ4', true, description: 'Is liblz4 present?')
+endif
+
+# Find required openssl library
+lib_openssl = dependency('openssl')
+
+# Find required pq library
+lib_pq = dependency('libpq')
+
+# Find required xml library
+lib_xml = dependency('libxml-2.0')
+
+# Find required yaml library (only used for build)
+lib_yaml = meson.get_compiler('c').find_library('yaml')
```

Comment:
There is canonically a yaml-0.1.pc, so you should use `dependency('yaml-0.1')` here.

You can also use `meson wrap install libyaml` to install the Meson wrap for this as a fallback.

Reply:
Thanks, fixed in 25f9a03e0c90. I've had surprisingly little luck finding examples for specific libs.